### ISSUE TITLE
Unexpected `Set for IN is not created yet` error

### ISSUE THREAD

**Issue by alesapin** [ClickHouse/ClickHouse#8859]
**How to reproduce**
```
CREATE TABLE testmt
(
    `CounterID` UInt64, 
    `value` String
)
ENGINE = MergeTree()
ORDER BY CounterID;

INSERT INTO testmt VALUES (1, '1');

SELECT arrayJoin([CounterID NOT IN (2)]) AS counter
FROM testmt
WHERE CounterID IN (2)
GROUP BY counter;

Code: 49. DB::Exception: Received from localhost:9000. DB::Exception: Set for IN is not created yet.
```
Introduced in https://github.com/ClickHouse/ClickHouse/pull/7497.

**Comment by alesapin** [comment]
@amosbird Maybe you have any ideas about how to fix it?

**Comment by amosbird** [comment]
Interesting.

a reduced test:

`select CounterID NOT IN (2) a FROM testmt WHERE CounterID IN (2) group by a;`

however these two variants work:

```
SELECT CounterID NOT IN (3) a FROM testmt WHERE CounterID IN (2) group by a;

SELECT CounterID NOT IN (2) a FROM testmt WHERE CounterID IN (2);
```

Is it somehow urgent? I might no be available recently.

**Comment by filimonov** [comment]
Fixed in #8868
